### DISCUSSION TITLE
Infiniate scrollable tables

### DISCUSSION THREAD

**Discussion by beams** [livewire/flux#770] [Feature Requests]
To be honest I feel there is absolutely no need for a modern framework to provide table pagination. Pagination is really a very poor UX solution and was only ever created back when ajax requests were not possible.

It would be really nice to have a decent infinitely scrollable table with searching, filtering and sorting. If this was well constructed you could deprecate the paging entirely and focus on one modern component.

**Comment by ju5t** [comment]
> To be honest I feel there is absolutely no need for a modern framework to provide table pagination.

This very much depends on the use-case. I disagree that the pagination component should be deprecated.

> It would be really nice to have a decent infinitely scrollable table with searching, filtering and sorting. If this was well constructed you could deprecate the paging entirely and focus on one modern component.

Flux only provides a component to display paginated results; it doesn't provide the functionality. That's all Laravel. I think as soon as it touches backend functionality, which infinite scrolling requires, it shouldn't be a Flux component.

**Reply by joshhanley** [reply]
Yeah I personally don't like the UX of infinite scroll tables, I much prefer paginated tables.

Infinite scroll can be tough with Livewire as the payload can increase on each "load more". Not sure how best to handle that (or if it's a Flux component at all). Will be up to Caleb to decide if he wants to build something to support it though 🙂

**Reply by beams** [reply]
Firstly, in hindsight, I should perhaps have not raised 2 topics in one post (1: pagination vs infinite scroll, 2: need for infinite scroll as a component in flux) but they are somewhat related.

For the sake of simplifying things here, I want to focus on the suggestion that Flux should offer an infinite scroll table solution. 

I use them in every project I build. I can't think of a single project that has not required one. The tables sometimes load hundreds of thousands of rows and a well-built lazy-loading infinite scroll is by far the most efficient user-friendly solution. So I find it to be an absolutely indispensable component in every project I work on.

So I do think Flux should offer an infinite scroll table as a core component.

I appreciate that this may be complicated by the need for specific backend functionality but I think that is not necessarily a reason to give up on the idea. Flux has demonstrated, in only a short time, such a dedication to high-quality UX and I think the lack of an infinite scroll table is one gaping hole in the project that would be great to fill.

**Comment by aurawindsurfing** [comment]
@beams you can have infinite scroll tables in flux already. I use below with Alpine plugin intersect and it works well with pagination:

```blade
    <div x-intersect.margin.50%="$wire.loadMore()">
        <flux:icon.loading class="hidden" wire:loading.class.remove="hidden"></flux:icon.loading>
    </div>
```

```php
    public function loadMore(): void
    {
        $this->limit += $this->perPage;
    }
```

```php
Model::all()->paginate($this->limit);
```
